src/unittests/f32_interval.spec.ts

```diff
@@ -300,134 +304,990 @@ g.test('contains_interval')
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+      ],
+      expected: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+      ],
+    },
+    {
+      input: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+        [9, 10, 11, 12],
+      ],
+      expected: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+        [9, 10, 11, 12],
+      ],
+    },
+    {
+      input: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+        [9, 10, 11, 12],
+        [13, 14, 15, 16],
+      ],
+      expected: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+        [9, 10, 11, 12],
+        [13, 14, 15, 16],
+      ],
+    },
+
+    // IntervalBounds
+    {
+      input: [
+        [[1], [2]],
+        [[3], [4]],
+      ],
+      expected: [
+        [1, 2],
+        [3, 4],
+      ],
+    },
+    {
+      input: [
+        [[1], [2]],
+        [[3], [4]],
+        [[5], [6]],
+      ],
+      expected: [
+        [1, 2],
+        [3, 4],
+        [5, 6],
+      ],
+    },
+    {
+      input: [
+        [[1], [2]],
+        [[3], [4]],
+        [[5], [6]],
+        [[7], [8]],
+      ],
+      expected: [
+        [1, 2],
+        [3, 4],
+        [5, 6],
+        [7, 8],
+      ],
+    },
+    {
+      input: [
+        [[1], [2], [3]],
+        [[4], [5], [6]],
+      ],
+      expected: [
+        [1, 2, 3],
+        [4, 5, 6],
+      ],
+    },
+    {
+      input: [
+        [[1], [2], [3]],
+        [[4], [5], [6]],
+        [[7], [8], [9]],
+      ],
+      expected: [
+        [1, 2, 3],
+        [4, 5, 6],
+        [7, 8, 9],
+      ],
+    },
+    {
+      input: [
+        [[1], [2], [3]],
+        [[4], [5], [6]],
+        [[7], [8], [9]],
+        [[10], [11], [12]],
+      ],
+      expected: [
+        [1, 2, 3],
+        [4, 5, 6],
+        [7, 8, 9],
+        [10, 11, 12],
+      ],
+    },
+    {
+      input: [
+        [[1], [2], [3], [4]],
+        [[5], [6], [7], [8]],
+      ],
+      expected: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+      ],
+    },
+    {
+      input: [
+        [[1], [2], [3], [4]],
+        [[5], [6], [7], [8]],
+        [[9], [10], [11], [12]],
+      ],
+      expected: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+        [9, 10, 11, 12],
+      ],
+    },
+    {
+      input: [
+        [[1], [2], [3], [4]],
+        [[5], [6], [7], [8]],
+        [[9], [10], [11], [12]],
+        [[13], [14], [15], [16]],
+      ],
+      expected: [
+        [1, 2, 3, 4],
+        [5, 6, 7, 8],
+        [9, 10, 11, 12],
+        [13, 14, 15, 16],
+      ],
+    },
 
-      // Double Intervals
-      { intervals: [[0, 1], [2, 5]], expected: [0, 5]},
-      { intervals: [[2, 5], [0, 1]], expected: [0, 5]},
-      { intervals: [[0, 2], [1, 5]], expected: [0, 5]},
-      { intervals: [[0, 5], [1, 2]], expected: [0, 5]},
-      { intervals: [[kValue.f32.infinity.negative, 0], [0, kValue.f32.infinity.positive]], expected: kAny},
+    // F32Interval
+    {
+      input: [
+        [toF32Interval(1), toF32Interval(2)],
```

Comment:
Would be nice if some of these provided something other than a single `number` to `toF32Interval`